Set up grid layout for multiple listings (#88)

lhlee-umass opened the pull request:
- Implement a dynamic container that adjusts based on screen size
- Use TailwindCSS Grid to create layout
- Layout should include reuse of reusable listing card component
- Appropriate / agreed upon header title for the layout
- UI tested for filtering
- Text color for tags changed to black to make it readable

This PR will result in finishing issues:
https://github.com/lhlee-umass/streetside/issues/88
https://github.com/lhlee-umass/streetside/issues/90

lrdiaz committed:
Updated JSX code to TSX

lrdiaz committed:
Created Card.tsx as reusable ListingCard template

lrdiaz committed:
Moved Card.tsx content to listing card

lrdiaz committed:
Deleted Card.tsx

angho8 committed:
Create filter-sorting.js

angho8 committed:
filtering and sorting

lauriehlee committed:
layout: set up grid layout for multiple listings

- Implement a dynamic container that adjusts based on screen size
- Use CSS Grid to create layout
- Layout should include reuse of reusable listing card component
- Appropriate / agreed upon header title for the layout

lauriehlee committed:
layout: set up grid layout for multiple listings

- Implement a dynamic container that adjusts based on screen size
- Use CSS Grid to create layout
- Layout should include reuse of reusable listing card component
- Appropriate / agreed upon header title for the layout

angho8 committed:
fixes/updates

lrdiaz committed:
Added .DS_store to .gitignore

zhandavidz committed:
fix: Remove DS_Store, update gitignore and package.json

- Removed .DS_Store
- Put .DS_Store into correct .gitignore
- Updated package.json to put the dependencies in the correct package.json

Helps resolve #98 and #99

zhandavidz committed:
Merge pull request #99 from lhlee-umass/98-add-to-gitignore

- removed `.DS_store` + added to `.gitignore` #98
- updates formerly JSX code to TSX for compatibility with rest of code, originally from PR #96

Addresses issues #89 and #94

angho8 commented:
Looks good, ensuring the layout is fully responsive and integrates the reusable listing card component will help maintain consistency.

zhandavidz committed:
fix: Fix some config file issues

- updated tsconfig.json to be more uniform with rest of configs
- updated package.json to use existing dependencies

Resolves PR #95

zhandavidz committed:
misc: Format the code with ESLint and Prettier

- ran `npm run format` to automatically fix ESLint and Prettier issues

Resolves PR #95

zhandavidz committed:
Merge pull request #95 from lhlee-umass/Filter/Sorting

- marketplace filtering/sorting

lauriehlee committed:
layout: multiple listings w/ mock data & filtering

- TailwindCSS grid Layout of multiple listings w/ mock data
- Checked the following:
- You can view all the desired attributes of the listing
- You can see a filtering option that suggests you can filter on category and price range
- You can see a sorting option that suggests you can sort based on category and price

lauriehlee committed:
Merge branch 'layout/88-frontend' of https://github.com/lhlee-umass/streetside into layout/88-frontend

lauriehlee committed:
Fixed linting issue

- Linting issue did not allow for merge, so it was fixed

zhandavidz reviewed (approved):
Everything looks good, great work!!! LGTM
